crates/vhost-user-backend/src/backend.rs

```diff
@@ -384,6 +473,21 @@ impl<T: VhostUserBackendMut> VhostUserBackend for RwLock<T> {
             .unwrap()
             .handle_event(device_event, evset, vrings, thread_id)
     }
+
+    fn set_device_state_fd(
+        &self,
+        direction: VhostTransferStateDirection,
+        phase: VhostTransferStatePhase,
+        file: File,
+    ) -> Result<Option<File>> {
+        self.write()
+            .unwrap()
+            .set_device_state_fd(direction, phase, file)
+    }
+
+    fn check_device_state(&self) -> Result<()> {
+        self.write().unwrap().check_device_state()
```

Comment:
Again my fault (like the missing comment), but won’t `read()` suffice here? `VhostUserBackendMut::check_device_state()` only takes an immutable `&self`.

Reply:
Yes, I missed that one